std/array.d

```diff
@@ -3339,6 +3340,69 @@ if (isDynamicArray!A)
     app.put(r[]);
 }
 
+unittest // Issue 13300
+{ 
+    // Issue 13300: appender must work with non pure postblit
+    import std.array : array;
+
+    struct Simple13300
```

Comment:
I don't think that it makes much difference in this case, but any struct or class declared inside a function which isn't declared as `static` has access to the function's scope and can sometimes screw with tests. So, it would be better to marke each of these structs as `static`.